Add swagger spec for new resource type in Microsoft.PolicyInsights

eladperets opened the pull request:
This commit adds a spec of a new preview resource type for the Microsoft.PolicyInsights- PolicyTrackedResources.
The new resource type is only avaliable in the 2018-07-01-preview API version. It only supports POSTing query (like other resource types in Microsoft.PolicyInsights).
The API itself is already deployed and passed an initial review by the ARM team.
If required, I can provide a short document that describes the API in more detail.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [V] The title of the PR is clear and informative.
- [V] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [V ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [V] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [V] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [V] My spec meets the review criteria:
  - [V] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [V] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [V] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

eladperets committed:
Add swagger spec for new resource type in Microsoft.PolicyInsights

This commit adds a spec of a new preview resource type for the Microsoft.PolicyInsights- PolicyTrackedResources.
The new resource type is only avaliable in the 2018-07-01-preview API version.
The API itself is already deployed and passed an initial review by the ARM team.

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-node

dsgouda reviewed (changes requested):
Please create a new tag in the config file (readme.md) that references this version
This enables the linter in CI to run against these specs

AutorestCI commented:
# Automation for azure-sdk-for-java

eladperets committed:
Add operations API to the spec and update readme.md

dsgouda reviewed (changes requested):
Looks OK apart from a few comments

dsgouda commented on specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyTrackedResources.json:
```diff
@@ -0,0 +1,526 @@
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PolicyInsights/policyTrackedResources/{policyTrackedResourcesResource}/queryResults": {
+      "post": {
+        "operationId": "PolicyTrackedResources_ListQueryResultsForResourceGroup",
```
If you plan to release a new API version, the SDK versions will be bumped to major version anyway, I will leave this to your discretion.
If the operation is not pageable, you can simply remove the `x-ms-pageable` extension, if it is you can request a suppression for the violation (I guess it would be valid in this case)

eladperets replied:
I prefer to keep this convention (it's the same in the other specs for our RP)

dsgouda commented on specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyTrackedResources.json:
```diff
@@ -0,0 +1,526 @@
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
```
It's a standard guideline we enforce for pageable operations

eladperets replied:
got it, added readonly properties

eladperets commented on specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyTrackedResources.json:
```diff
@@ -0,0 +1,526 @@
+        "operationId": "PolicyTrackedResources_ListQueryResultsForSubscription",
+        "description": "Queries policy tracked resources under the subscription.",
+        "parameters": [
+          {
```
I sampled some random RPs and notices they actually added those at the end of the section.

dsgouda replied:
Either at the beginning or at the end works, just keep them together

eladperets committed:
Minor fixes

- Moved subscriptionId and apiVersion parameters to the end of the
sections.
- Removed the 'required' section from the definitions (irrelevant since
the API doesn't support PUT operations).
- Marked readonly properties.

dsgouda reviewed (approved):
LGTM
